LifeCycleAssessment_Engine/Compute/EvaluateEnvironmentalProductDeclaration.cs

```diff
@@ -0,0 +1,194 @@
+        [Output("quantity", "The quantity of the desired metric provided by the EnvironmentalProductDeclarationField")]
+        public static double EvaluateEnvironmentalProductDeclarationPerObject(IBHoMObject obj, EnvironmentalProductDeclaration environmentalProductDeclaration, EnvironmentalProductDeclarationField environmentalProductDeclarationField = EnvironmentalProductDeclarationField.GlobalWarmingPotential) //default to globalWarmingPotential evaluation
+        {
+            if (obj != null)
+            {
+                //First check validity of inputs
+                object vol = obj.PropertyValue("Volume");
+                if (vol == null)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("The input object must have a Volume property for this method to work.");
+                    return 0;
+                }
+                double volume = System.Convert.ToDouble(vol);
+                if (volume == 0)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("The input object's Volume value is invalid. Volume should be in m3 in numerical format.");
+                    return 0;
+                }
+
+                //Check the epd declared unit type to determine whether to calculate by mass, volume, or neither
+                string declaredUnitType = environmentalProductDeclaration.DeclaredUnitType();
+
+                if (declaredUnitType == "Volume")
+                {
+                    return EvaluateEnvironmentalProductDeclarationByVolume(environmentalProductDeclaration, environmentalProductDeclarationField, volume);
+                }
+
+                else if (declaredUnitType == "Mass")
+                {
+                    double density = 0;
+                    string epdDensity = environmentalProductDeclaration.Density.ToString() ?? "";
+                    double densityVal = System.Convert.ToDouble(epdDensity.Substring(0, epdDensity.IndexOf(" ")));
+                    if (System.Convert.ToDouble(densityVal) != 0)
+                    {
+                        density = densityVal;
+                        BH.Engine.Reflection.Compute.RecordNote(String.Format("This method is using a density of {0} supplied by the EnvironmentalProductDeclaration to calculate Mass.", density));
+                    }
+                    else if (obj.PropertyValue("Density") == null)
+                    {
+                        BH.Engine.Reflection.Compute.RecordNote("The EnvironmentalProductDeclaration and input object do not have density information. The epd is mass-based. Please add density data to the input object.");
+                        return 0;
+                    }
+                    else if (System.Convert.ToDouble(obj.PropertyValue("Density")) == 0)
+                    {
+                        BH.Engine.Reflection.Compute.RecordNote("The input object's Density value is invalid. Density should be in kg/m3 in numerical format.");
```

Comment:
Same as previous comment, can we accept `density` being less than 0? If not, I'd advise we change the `if` condition and note accordingly.